components/SegmentedControl/src/EqualPanel.cs

```diff
@@ -58,32 +70,24 @@ protected override Size MeasureOverride(Size availableSize)
 
     protected override Size ArrangeOverride(Size finalSize)
     {
-        var x = 0.0;
-        foreach (var child in Children)
-        {
-            child.Arrange(new Rect(x, 0, maxItemWidth, maxItemHeight));
-            x += maxItemWidth + Spacing;
-        }
-        return finalSize;
-    }
+        double newWidth = 0.0;
 
-    private void SetMaxDimensions(UIElementCollection items, Size availableSize)
-    {
-        foreach (var item in items)
+        foreach (var child in Children)
         {
-            item.Measure(availableSize);
-            double desiredWidth = item.DesiredSize.Width;
-            if (desiredWidth > maxItemWidth)
+            if (child.Visibility != Visibility.Collapsed)
             {
-                maxItemWidth = desiredWidth;
+                child.Arrange(new Rect(newWidth, 0, maxItemWidth, maxItemHeight));
+                newWidth += maxItemWidth + Spacing;
             }
+        }
 
-            double desiredHeight = item.DesiredSize.Height;
-            if (desiredHeight > maxItemHeight)
-            {
-                maxItemHeight = desiredHeight;
-            }
+        // Check if there's more width available - if so, recalculate (e.g. whenever Grid.Column is set to Auto)
+        if (finalSize.Width > newWidth)
+        {
+            MeasureOverride(finalSize);
         }
```

Comment:
We can calculate the `newWidth` based on the Count alone (well the uncollapsed count, we could cache that number or this result in the measure step).

As if we're going to re-measure everything if we have more space, then we should want to do that before we tell everyone what their final size is with `child.Arrange`, no?